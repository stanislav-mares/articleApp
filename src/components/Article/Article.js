import React from 'react';
import PropTypes from 'prop-types';

const Article = (props) => {

  return(
      <div>
        <h2>Article:</h2>
        {props.text && (props.text.length > 0) ? <p>{props.text}</p> : 'No text'}
        {props.author && (props.author.length > 0) ?
          <h4>{`${props.author}, ${new Date(props.date).toLocaleString()}`}</h4> : 'No author'}
        <hr />
      </div>
    );
}

Article.propTypes = {
  text: PropTypes.string,
  autor: PropTypes.string,
  date: PropTypes.string
};

export default Article;
