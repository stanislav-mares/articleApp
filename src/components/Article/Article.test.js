import React from 'react';
import enzymeConfig from 'enzymeConfig/enzymeConfig';
import { shallow } from 'enzyme';

import Article from 'components/Article/Article';

describe('<Article />', () => {

  const article = {
    author: 'Eliana Franco',
    date: '2017-09-10T22:00:05.919Z',
    text: `Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.`
  };

  const wrongArticle1 = {
    author: 'Eliana Franco',
    date: '2017-09-10T22:00:05.919Z'
  };

  const wrongArticle2 = {
    author: 'Eliana Franco',
    date: '2017-09-10T22:00:05.919Z',
    text: ''
  };

  const wrongArticle3 = {
    date: '2017-09-10T22:00:05.919Z',
    text: `Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.`
  };

  const wrongArticle4 = {
    author: '',
    date: '2017-09-10T22:00:05.919Z',
    text: `Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.`
  };

  it('Testing if no text is rendered when no text is passed', () => {
    const component = shallow(<Article { ...wrongArticle1 }/>);

    expect(component.find('p').length).toBe(0);
  });

  it('Testing if no text is rendered when empty text is passed', () => {
    const component = shallow(<Article { ...wrongArticle2 }/>);

    expect(component.find('p').length).toBe(0);
  });

  it('Testing if no author and date is rendered when no author is passed', () => {
    const component = shallow(<Article { ...wrongArticle3 }/>);

    expect(component.find('h4').length).toBe(0);
  });

  it('Testing if no author and date is rendered when empty author is passed', () => {
    const component = shallow(<Article { ...wrongArticle4 }/>);

    expect(component.find('h4').length).toBe(0);
  });

  it('Testing structure of the Article component', () => {
    const component = shallow(<Article { ...article }/>);

    expect(component).toMatchSnapshot();
  });
});
