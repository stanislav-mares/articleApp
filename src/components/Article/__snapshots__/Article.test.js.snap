// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> Testing structure of the Article component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article
    author="Eliana Franco"
    date="2017-09-10T22:00:05.919Z"
    text="Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Article:
        </h2>,
        <p>
          Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.
        </p>,
        <h4>
          Eliana Franco, 2017-9-11 00:00:05
        </h4>,
        <hr />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Article:",
        },
        "ref": null,
        "rendered": "Article:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.",
        },
        "ref": null,
        "rendered": "Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Eliana Franco, 2017-9-11 00:00:05",
        },
        "ref": null,
        "rendered": "Eliana Franco, 2017-9-11 00:00:05",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Article:
          </h2>,
          <p>
            Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.
          </p>,
          <h4>
            Eliana Franco, 2017-9-11 00:00:05
          </h4>,
          <hr />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Article:",
          },
          "ref": null,
          "rendered": "Article:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.",
          },
          "ref": null,
          "rendered": "Your bones don't break, mine do. That's clear. Your cells react to bacteria and viruses differently than mine. You don't get sick, I do. That's also clear. But for some reason, you and I react the exact same way to water. We swallow it too fast, we choke. We get some in our lungs, we drown. However unreal it may seem, we are connected, you and I. We're on the same curve, just on opposite ends.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Eliana Franco, 2017-9-11 00:00:05",
          },
          "ref": null,
          "rendered": "Eliana Franco, 2017-9-11 00:00:05",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
